mapie/conformity_scores/conformity_scores.py

```diff
@@ -158,20 +184,148 @@ def get_conformity_scores(
 
         y_pred: NDArray
-            Predicted values.
+            Predicted target values.
 
         Returns
         -------
         NDArray
             Conformity scores.
         """
-        conformity_scores = self.get_signed_conformity_scores(y, y_pred)
+        conformity_scores = self.get_signed_conformity_scores(X, y, y_pred)
         if self.consistency_check:
-            self.check_consistency(y, y_pred, conformity_scores)
+            self.check_consistency(X, y, y_pred, conformity_scores)
         if self.sym:
             conformity_scores = np.abs(conformity_scores)
         return conformity_scores
+
+    @staticmethod
+    def get_quantile(
+        values: NDArray,
+        alpha_np: NDArray,
+        axis: int,
+        method: str
+    ) -> NDArray:
+        """
+        Compute the alpha quantile of the conformity scores considering
+        the symmetrical property if so.
+
+        Parameters
+        ----------
+        values: NDArray
+            Values from which the quantile is computed, it can be the
+            conformity scores or the conformity scores aggregated with
+            the predictions.
+
+        alpha_np: NDArray
+            NDArray of floats between ``0`` and ``1``, represents the
+            uncertainty of the confidence interval.
+
+        axis: int
+            The axis from which to compute the quantile.
+
+        method: str
+            ``"higher"`` or ``"lower"`` the method to compute the quantile.
+
+        Returns
+        -------
+        NDArray
+            Lower and upper quantile of the prediction intervals.
+            These quantiles are identical if the score is not symmetrical.
+        """
+        quantile = np.column_stack([
+            np_nanquantile(
+                values.astype(float),
+                _alpha,
+                axis=axis,
+                method=method
+            )
+            for _alpha in alpha_np
+        ])
+        return quantile
+
+    def get_bounds(
+        self,
+        X: ArrayLike,
+        estimator: RegressorMixin,
+        conformity_scores: NDArray,
+        alpha_np: NDArray,
+        ensemble: bool,
+        method: str
+    ) -> Tuple[NDArray, NDArray, NDArray]:
+        """
+        Compute bounds of the prediction intervals from the observed values,
+        the estimator of MapieRegressor and the conformity scores.
+
+        Parameters
+        ----------
+        X: ArrayLike
+            Observed feature values.
+
+        estimator: RegressorMixin
+            Estimator that is fitted to predict y from X.
+
+        conformity_scores: ArrayLike
+            Conformity scores.
+
+        alpha_np: NDArray
+            NDArray of floats between ``0`` and ``1``, represents the
+            uncertainty of the confidence interval.
+
+        ensemble: bool
+            Boolean determining whether the predictions are ensembled or not.
+
+        method: str
+            The method parameter of MapieRegressor.
```

Comment:
UPD: add more detail to this description, specifying the different values that `method` can take (`base, minmax, plus`). It's better not to talk about `MapieRegressor` here.

Reply:
Don't confuse usage (the goal) with definition (the way). Yes, I agree with you that we're using the `MapieRegressor` method parameter in practice, but `ConformityScore` is a class that should be self-sufficient.

Reply:
Should I add in this docstring the same description as the one in MapieRegressor  ? : ` Method to choose for prediction interval estimates.
        Choose among:

        - ``"naive"``, based on training set conformity scores,
        - ``"base"``, based on validation sets conformity scores,
        - ``"plus"``, based on validation conformity scores and
          testing predictions,
        - ``"minmax"``, based on validation conformity scores and
          testing predictions (min/max among cross-validation clones).`

Reply:
Method to choose for prediction interval estimates. The "plus" method implies that the quantile is calculated after estimating the bounds, whereas the other methods (among the ``"naive"``, ``"base"`` or ``"minmax"`` methods, for example) do the opposite.

Reply:
Check that this proposition is correct before copy/paste.